drop-transfer/src/manager.rs

```diff
@@ -635,16 +646,33 @@ impl TransferManager {
         state.ensure_not_terminated()
     }
 
-    pub async fn outgoing_remove(&self, transfer_id: Uuid) -> crate::Result<()> {
+    /// Returns `None` if the cancel event should be suppressed
```

Comment:
hmm... I think this function is a bit too complex. I dislike that it knows something about the canceling and events, it should basically judging from the name - validate and remove. Maybe return what it removed.

I understand that it has multiple call sites so moving the logic into there is not scalable. Maybe you have ideas how to improve on it?

Reply:
Um, it's 13 lines of code, not that complex IMHO. It does remove the transfer if it exists and returns leftovers when appropriate.
I was thinking about emitting events here but I deliberately try to avoid emitting events from the manager because its calls are mutex-protected and I'm in fear of creating accidental deadlocks or something like that. Another reason is that I want to decouple event emissions from the manager for clarity.
However, these kinds of "hacks" are necessary to communicate to the caller what happened.
So, I sort of agree about the naming but I don't see how this can be solved better

Reply:
I am not particularly unhappy about amount of LoC but rather about it's responsibility, that it does something extra, like checks the state and returns an Option. If you also don't see a way to simplify it then it's fine